### PR TITLE
(Capacitor): Add migration guide for new version

### PR DESCRIPTION
**Pull request by lucas-zimerman**
## DESCRIBE YOUR PR
The next major version of Sentry Capacitor will also use Sentry Javascript SDK V8 where both includes break changes, this PR aims to add a migration guide for both Capacitor and the Sibling SDKs.

I also removed the deprecated setup of Capacitor 2 that we will no longer support and also a small refactor on v8-new-performance-api to make the file smaller to read on the code review.

## IS YOUR CHANGE URGENT?  

Help us prioritize incoming PRs by letting us know when the change needs to go live.
- [ ] Urgent deadline (GA date, etc.): 
- [ ] Other deadline: 
- [X] None: Not urgent, can wait up to 1 week+

## SLA

- Teamwork makes the dream work, so please add a reviewer to your PRs.
- Please give the docs team up to 1 week to review your PR unless you've added an urgent due date to it.
Thanks in advance for your help!

## PRE-MERGE CHECKLIST

I checked that locally it is building and displaying correctly.

*Make sure you've checked the following before merging your changes:*

- [ ] Checked Vercel preview for correctness, including links
- [ ] PR was reviewed and approved by any necessary SMEs (subject matter experts)
- [ ] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## EXTRA RESOURCES

- [Sentry Docs contributor guide](https://docs.sentry.io/contributing/)

### CODE REVIEW

**Comment by lucas-zimerman** [comment]
@krystofwoldrich  @kahest  

Should I also add an information like this? https://docs.sentry.io/platforms/javascript/guides/angular/#angular-version-compatibility

despite the page mentioning that older versions of Angular should use V7, the docs only only shows the documentation of V8 so it may generate confusion to users sticking with V7.

Our Capacitor SDK only supports Angular as low as 10, so we could mention that users should stick with Capacitor V0 for Angular 10 to - 13 and V1 for Angular 14 or higher.

**Comment by kahest** [comment]
@chargome I added @getsentry/team-web-sdk-frontend as reviewer - please rope in other reviewers if needed 🙏

**Comment by kahest** [comment]
> @krystofwoldrich @kahest
> 
> Should I also add an information like this? [docs.sentry.io/platforms/javascript/guides/angular#angular-version-compatibility](https://docs.sentry.io/platforms/javascript/guides/angular/#angular-version-compatibility)
> 
> despite the page mentioning that older versions of Angular should use V7, the docs only only shows the documentation of V8 so it may generate confusion to users sticking with V7.
> 
> Our Capacitor SDK only supports Angular as low as 10, so we could mention that users should stick with Capacitor V0 for Angular 10 to - 13 and V1 for Angular 14 or higher.

Yes I think that would help users understand 👍

**Comment by krystofwoldrich** [comment]
Besides the Angular versions mentioned above, the guide migration guide looks good!

File: docs/platforms/javascript/common/migration/v7-to-v8/v8-new-performance-api.mdx [hunk 1]
```diff
@@ -162,46 +162,7 @@ Previously, spans & transactions had a bunch of properties and methods to be use
...
-| --------------------------- | ------------------------------------------------ |
-| `name`                      | `spanToJSON(span).description`                   |
-| `trimEnd`                   | Removed                                          |
-| `parentSampled`             | `spanIsSampled(span)` & `spanContext().isRemote` |
-| `metadata`                  | Use attributes instead or set on scope           |
-| `setContext()`              | Set context on scope instead                     |
-| `setMeasurement()`          | `Sentry.setMeasurement()`                        |
-| `setMetadata()`             | Use attributes instead or set on scope           |
-| `getDynamicSamplingContext` | `getDynamicSamplingContextFromSpan(span)`        |
+<Include name="migration/javascript-v8/new-span-schema" />
```
Comment (lforst) [inline docs/platforms/javascript/common/migration/v7-to-v8/v8-new-performance-api.mdx:165]: Is there a reason we pulled this out? If not I would inline this again.

**Review by lforst — COMMENTED** [review]
Thanks!

**Review by krystofwoldrich — APPROVED** [review]
Looks good! Thank you! 🚀

**Pull request merged**